central/virtualmachine/service/service_impl.go

```diff
@@ -76,6 +94,25 @@ func (s *serviceImpl) ListVirtualMachines(ctx context.Context, request *v2.ListV
 	for _, vm := range vms {
 		v2VMs = append(v2VMs, storagetov2.VirtualMachine(vm))
 	}
+	requestQueryPagination := request.GetQuery().GetPagination()
+	if requestQueryPagination.GetSortOption() == nil && len(requestQueryPagination.GetSortOptions()) == 0 {
```

Comment:
```suggestion
	if len(requestQueryPagination.GetSortOptions()) == 0 {
```
Isn't the first part redundant since nil slices have a length of zero?

Reply:
These two conditions target different fields in the proto. The naming is a bit unfortunate and can lead to confusion as it differs only by the trailing s `GetSortOption` (single option) vs `GetSortOptions` (array of options).